### PR TITLE
Animation refactor

### PR DESCRIPTION
**Pull request by trvrb**
Progress on #367 

Before this branch (current `master`) if getting the following performance for Ebola:

* ~230ms animation frames
* ~20ms `getLatLongs`

With this branch in place I'm getting:

* ~190ms animation frames
* ~10ms `getLatLongs`
* ~50ms `updateVisibility`

Before this branch (current `master`) if getting the following performance for 3y flu region resolution:

* ~400ms animation frames
* ~45ms `getLatLongs`

With this branch in place I'm getting:

* ~270ms animation frames
* ~15ms `getLatLongs`
* ~150ms `updateVisibility`

_Going to keep updating these numbers as I work on the PR. Serving to help me remember if there's performance increases._

### CODE REVIEW

**Comment by trvrb** [comment]
Okay. I think I'm done with this for the moment. @colinmegill: I tried to clean up code where possible, things like:

* replacing things like `long0` with `longOrig`, `long1` with `longDest` and `winningPair` with `closestPair` to make self documenting
* don't pass around high-level redux object `controls` to things like `updateVisibility` that have to be reached into and instead pass in low-level `numDates`.
* carve out functions for `leafletLatLongToLayerPoint` and `maybeGetTransmissionPair` to reduce code duplication

Biggest functional change is duplicating map with adding a half map to the west and a half map to the east, rather than the full triplicating. This improves performance and provides a more natural set of bounds.

This is just FYI for @colinmegill, but @jameshadfield could you do a quick review on Monday and merge if it looks good.

**Comment by jameshadfield** [comment]
Looks good to me - i've made some comments in issue #367

Will let @colinmegill take a look, but I plan to merge later this afternoon (since animation is currently disabled in master anyway)

**Comment by colinmegill** [comment]
Ok, will take a look now!

**Comment by trvrb** [comment]
Great. Thanks both.

File: src/components/map/map.js [hunk 1]
```diff
@@ -488,38 +497,6 @@ class Map extends React.Component {
       }
     }, animationTick);
 
-    // controls: state.controls,
```
Comment (colinmegill) [inline src/components/map/map.js]: This comment can stay, I will clean up on the requestAnimationFrame pass

File: src/components/map/map.js [hunk 2]
```diff
@@ -25,18 +24,16 @@ import {
     // datasetGuid: state.tree.datasetGuid,
     treeVersion: state.tree.version,
     treeLoaded: state.tree.loaded,
-    controls: state.controls,
+    splitTreeAndMap: state.controls.splitTreeAndMap,
```
Comment (colinmegill) [inline src/components/map/map.js]: This is way cleaner

File: src/util/mapHelpersLatLong.js [hunk 3]
```diff
@@ -94,134 +122,65 @@ export const getLatLongs = (nodes, visibility, metadata, map, colorBy, geoResolu
...
-        )
-      });
+      let lat = geo[geoResolution][key].latitude;
+      let long = geo[geoResolution][key].longitude + OFFSET;
+      if (long > westBound && long < eastBound) {
+        demesAndTransmissions.demes.push({
+          location: key, // Thailand:
+          total: value.length, // 20, this is an array of all demes of a certain type
+          color: averageColors(value),
+          coords: leafletLatLongToLayerPoint(lat, long, map)
```
Comment (colinmegill) [inline src/util/mapHelpersLatLong.js]: This (apparently, but I could be wrong here) no longer relies on `new L.LatLng`, is that intentional?

**Comment by colinmegill** [comment]
All set - this is looking great

**Comment by trvrb** [comment]
These changes look good James.

File: src/util/mapHelpersLatLong.js (hunk 3 above)
Comment (trvrb) [inline src/util/mapHelpersLatLong.js]: I did this to reduce code duplication. `leafletLatLongToLayerPoint` should be handing back a `new L.LatLng(lat, long)` in the exact same fashion as before, just shelved out into a function.

**Pull request merged**